pre-docs/writing_yaml_plans.md

```diff
@@ -473,3 +473,134 @@ plan mymodule::yamlplan(
 }
 ```
 
+## YAML plan constructs which cannot be translated to Puppet plans
+
+This section points out some of the quirks and limitations associated with converting a plan expressed as YAML to a Puppet plan. In some cases it is impossible to accurately translate from YAML to Puppet, in others code that is generated from the conversion is syntactically correct but not idiomatic Puppet. 
+
+### Named eval step
+
+The eval step allows snippets of Puppet code to be expressed in YAML plans. When converting a multi-line eval step to Puppet and storing the resulting value in a variable the `with` lambda is used. For example, the following YAML plan:
+```
+parameters:
+  foo:
+    type: Optional[Integer]
+    description: foo
+    default: 0
+
+steps:
+  - eval: |
+      $x = $foo + 1
+      $x * 2
+    name: eval_step
+
+return: $eval_step
+```
+Would be converted to:
+```
+plan yaml_plans::with_lambda(
+  Optional[Integer] $foo = 0
+) {
+  $eval_step = with() || {
+    $x = $foo + 1
+    $x * 2
+  }
+
+  return $eval_step
+}
+```
+If the author were writing the code as a Puppet plan they would likely not employ the use of the lambda. In this example the converted Puppet code is 'correct' but is not particularly natural or readable. 
```

Comment:
The quotation marks make it look like the correct code is actually incorrect. Recommend removing the quotations and shortening slightly:

>In this example the converted Puppet code is correct, but not natural or readable.